mypy/messages.py

```diff
@@ -1669,6 +1669,12 @@ def format(typ: Type) -> str:
     def format_list(types: Sequence[Type]) -> str:
         return ', '.join(format(typ) for typ in types)
 
+    def format_union(types: Sequence[Type]) -> str:
+        import mypy.options
+        if not mypy.options._based:
+            return format_list(types)
+        return ' | '.join(format(typ) for typ in types)
+
```

Comment:
this seems to be doing the same thing as https://github.com/KotlinIsland/basedmypy/pull/290/files#diff-265bbd9d8b179bfc6a75071cd473edbad7afbf73f011b830ac42c38438b60664R2559-R2568

can one be removed?

Reply:
I think `messages.format_type_inner` and `types.TypeStrVisitor` already duplicate everything and it looks like more work than it's work figuring out the import order etc, I think it's fine to leave this small function as is for now.